vlasovsolver/cpu_trans_pencils.hpp

```diff
@@ -97,6 +106,85 @@ struct setOfPencils {
       path.push_back(pathIn);
    }
 
+   void binPencils() {
+      binOfPencil.resize(N);
+
+      // Consider only cells which _any_ pencil writes into for binning,
+      // since read-only cells aren't affected by race conditions
+      std::unordered_set<CellID> targetCells = {};
+      #pragma omp parallel for
+      for (uint i = 0; i < sumOfLengths; ++i) {
+         const CellID targ = ids[i];
+         const Realf ratio = targetRatios[i];
+         if (targ && (ratio > 0.0)) {
+            #pragma omp critical
+            targetCells.insert(targ);
+         }
+      }
+
+      // Loop over pencils
+      // TODO could be paralellized as well
+      for (uint i = 0; i < N; ++i) {
+         binOfPencil[i] = i;
+         targetCellsInBin[i] = {};
```

Comment:
Let's please add a comment here clarifying what actually goes in this container. (and/or rename the container)

Reply:
Doxygen'd this, at least I think; not sure what a better name would be (just cellsInBin?)